server/src/main/java/org/elasticsearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -78,21 +79,27 @@ public Builder(String name, boolean hasDocValues) {
 
         @Override
         public BinaryFieldMapper build(BuilderContext context) {
-            return new BinaryFieldMapper(name, new BinaryFieldType(buildFullName(context), hasDocValues.getValue(), meta.getValue()),
-                    multiFieldsBuilder.build(this, context), copyTo.build(), this);
+            FieldType fieldType;
+            if (stored.getValue()) {
+                fieldType = StoredField.TYPE;
+            } else {
+                fieldType = new FieldType();
+            }
+            return new BinaryFieldMapper(name, new BinaryFieldType(buildFullName(context), hasDocValues.getValue(), fieldType,
+                meta.getValue()), multiFieldsBuilder.build(this, context), copyTo.build(), this);
         }
     }
 
     public static final TypeParser PARSER = new TypeParser((n, c) -> new Builder(n));
 
     public static final class BinaryFieldType extends MappedFieldType {
 
-        public BinaryFieldType(String name, boolean hasDocValues, Map<String, String> meta) {
-            super(name, false, hasDocValues, TextSearchInfo.NONE, meta);
```

Comment:
There's a few other places that use `TextSearchInfo.NONE` and return incorrect information from `isStored()` - we initially decided that we don't care, because it's only used for highlighting and none of the fields in question can be highlighted.

There's actually a TODO on `TextSearchInfo#isStored()` saying 'maybe move this directly onto MappedFieldType' which might be a better way of conveying this information?

Reply:
I think I remember now. This causes problems with binary field mapper, because it is not stored by default, hence we end up by default not adding it to _field_names, so the idea was to check if it's stored, and adjust the assertions, but I read the wrong value :)